mmv1/third_party/terraform/resources/resource_dataflow_flex_template_job.go.erb

```diff
@@ -27,6 +28,12 @@ func resourceDataflowFlexTemplateJob() *schema.Resource {
 		Read:   resourceDataflowFlexTemplateJobRead,
 		Update: resourceDataflowFlexTemplateJobUpdate,
 		Delete: resourceDataflowFlexTemplateJobDelete,
+		CustomizeDiff: customdiff.All(
+			resourceDataflowJobTypeCustomizeDiff,
```

Comment:
Does this resource have a batch type? If we need this `CustomizeDiff`, I think it would need to use its own function, and it could result in a breaking change.

Reply:
Yes, Dataflow Flex still has the same behaviour as Dataflow Jobs. It is just wrapper to run things from a container instead. But the target resource of  Dataflow job and Dataflow Flex is same and has same properties, I can copy the same function into dataflow flex as its own function. But It will be exactly same thing. The dataflow job options are less in terraform code because it is outdated. But I can say both dataflow job and flex have same options

Reply:
A few things here:
- Since a batch dataflow job can be used here, it does make sense to have this `CustomizeDiff` to make sure those jobs are always re-created when fields change
- Some fields are currently not ForceNew, and this change would make them ForceNew for batch jobs. We will need to make sure this is safe. In particular, we should check how they currently behave if a user tries to update them.
- Re-using the same CustomizeDiff function as dataflow jobs will probably not work because they use different schemas. `resourceDataflowJobTypeCustomizeDiff` uses `resourceDataflowJob().Schema`.